codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -318,6 +321,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
             // If this connection is closing and there are no longer any
             // active streams, close after the current operation completes.
             if (closeListener != null && connection().numActiveStreams() == 0) {
+                ChannelFutureListener closeListener = Http2ConnectionHandler.this.closeListener;
+                // This method could be reentrant, and we don't want to notify the closeListener multiple times
+                Http2ConnectionHandler.this.closeListener = null;
```

Comment:
> Let's just say that in the comment then? I think nulling the closeListener makes perfect sense, but the "reentrant" confused me.

@buchgr - Done.  Just trying to keep you on your toes :)  I think I was reviewing the other PR where I brought up "reentrant"...

@nmittler - You are right to question this...I don't think it needs to be `volatile`.  Upon further investigation `...the specified listener is notified immediately.` comment for the `addListener` method implicitly means that the notification is scheduled on the event loop (if not currently in the event loop).  So I will leave it as a regular variable :)